src/GovUk.Education.ExploreEducationStatistics.Common.Tests/Extensions/ServiceCollectionExtensions.cs

```diff
@@ -0,0 +1,87 @@
+using System;
+using System.Linq;
+using GovUk.Education.ExploreEducationStatistics.Common.ModelBinding;
+using Microsoft.EntityFrameworkCore;
+using Microsoft.Extensions.DependencyInjection;
+using Moq;
+using static Moq.MockBehavior;
+
+namespace GovUk.Education.ExploreEducationStatistics.Common.Tests.Extensions;
+
+/// <summary>
+/// This class contains a number of extension methods for IServiceCollection that are useful when setting up
+/// integration tests that require a customised Startup class.
+/// </summary>
+public static class ServiceCollectionExtensions
+{
+    /// <summary>
+    /// This method replaces a DcContext that has been registered in Startup with an in-memory equivalent.
+    /// </summary>
+    public static IServiceCollection ReplaceDbContext<TDbContext>(this IServiceCollection services)
```

Comment:
I suppose it was more about explaining that it expects to find an equivalent DbContext already registered that it will firstly remove and then replace with this instance.

Same for ReplaceService, but with the added addition there that it will be able to replace the existing registered Service with the same lifecycle as it was originally registered with.

I am however happy to rename these if you still want?

Reply:
I think my main issue is that the `ReplaceService` actually allows the consumer to provide their own implementation. This is inconsistent with a method like `ReplaceDbContext`, which doesn't allow this, despite being named very similarly.

Would suggest we go with either `UseInMemoryDbContext` or `UseTestDbContext`. We just might want to consider the naming based on potentially adding a proper database implementation in the future e.g. with Testcontainers.

Reply:
Fair enough.  I've changed to `UseInMemoryDbContext`, ta.